src/features/groups/list/group-details/index.tsx

```diff
@@ -0,0 +1,184 @@
+
+  const onValidationChange = useCallback((name: string, value: boolean) => {
+    setValidation(validation => ({
+      ...validation,
+      [name]: value,
+    }));
+  }, []);
+
+  const onFormSubmit = async () => {
+    if (form.id) {
+      updateGroup({
+        name: form.name,
+        currencyCode: form.currencyCode,
+        id: form.id
+      });
+    } else {
+      createGroup({
+        name: form.name,
+        currencyCode: form.currencyCode
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (createdGroup) {
+      onSaved(createdGroup);
+    }
+  }, [createdGroup, onSaved]);
+
+  useEffect(() => {
+    if (updatedGroup) {
+      onSaved(updatedGroup);
+    }
+  }, [updatedGroup, onSaved]);
+
+  useEffect(() => {
+    if (groupId) {
+      getGroupById(groupId).then(({ data }) => {
+        if (data) {
+          setForm(data);
+        }
+      });
+    }
+  }, [groupId, getGroupById]);
+
+  const onControlChange = useCallback((name: string, value: string) => {
+    setForm(form => ({
+      ...form,
+      [name]: value,
+    }));
+  }, []);
+
+  const onAddGroupMembersClick = useCallback(() => {
+    onGroupMembersModeChange(true);
+  }, [onGroupMembersModeChange]);
+
+  return (
+    <ContentWrapper fullWidth>
+      <GroupDetailsWrapper hidden={isAddGroupMembersMode} fullWidth>
+        <MainInfoWrapper fullWidth>
+          <GroupInfoWrapper gap={'2.5rem'} jc={'center'} fullWidth>
+            <Row jc={'center'} fullWidth>
+              <img src={addAvatar} alt={addAvatar} />
+            </Row>
+
+            <Row jc={'center'} fullWidth>
+              <Input
+                placeholder={t('groupName')}
+                reversedTheme
+                name={'name'}
+                required
+                value={form.name}
+                validator={requiredValidator}
+                onTextChange={onControlChange}
+                onValidationChange={onValidationChange}
+              />
+            </Row>
+
+            <Row jc={'space-between'} ai={'center'} fullWidth>
+              <GroupCurrencyText>
+                {t('mainCurrency')}
+              </GroupCurrencyText>
+
+              <Select
+                options={currencies}
+                required
+                value={form.currencyCode}
+                name={'currencyCode'}
+                reversedTheme={true}
+                onValueChanged={onControlChange}
+              />
+            </Row>
+          </GroupInfoWrapper>
+
+          <GroupDetailsBackground fullWidth>
+          </GroupDetailsBackground>
+        </MainInfoWrapper>
+
+        {
+          form.id &&
+          <AddMembersWrapper gap={'0.5rem'} jc={'center'} ai={'center'} fullWidth>
+            <TextUnderline onClick={onAddGroupMembersClick}>
+              {t('addGroupMembers')}
+            </TextUnderline>
+
+            <img src={arrowRight} alt={arrowRight} onClick={onAddGroupMembersClick} />
+          </AddMembersWrapper>
```

Comment:
I think it would be better not to hide this until the `getGroupById` request done.
It would be better to add additional check in `GroupMembers ` component (if the `getGroupById` request is done and we have `participants` data, and place some loader there if not)

You can trigger `getGroupById` request here. And also add 
```
const {data, isLoading} = useGetTransactionGroupByIdQuery();
```
in `GroupMembers ` component. This will not start new request but just gives you data from the request triggered here.